components/x-timeline-feed/src/TimelineFeed.jsx

```diff
@@ -0,0 +1,40 @@
+import { h } from '@financial-times/x-engine';
+import { Teaser, presets } from '@financial-times/x-teaser';
+import { getArticleGroups } from './lib/transform';
+import styles from './TimelineFeed.scss';
+import classNames from 'classnames';
+
+const TimelineFeed = props => {
+	const { articleActionsCreator = () => {} } = props;
+	const articleGroups = getArticleGroups(props);
+
+	return articleGroups.length && (
+		<div className={classNames(styles.root)}>
+			{articleGroups.map(group => (
+				<section key={group.date} className={classNames(styles.articleGroup)}>
+					<h2 className={classNames(styles.articleGroup__heading)}>{group.title}</h2>
+					<ul className={classNames(styles.articleGroup__articles)}>
+						{group.articles.map(article => {
+							const articleActions = articleActionsCreator(article);
+
+							return (
+								<li key={article.id} className={styles.article}>
+									<Teaser
+										{...article}
+										{...presets.SmallHeavy}
+									/>
+									{articleActions && <div
+										className={classNames(styles.articleActions)}
+										dangerouslySetInnerHTML={{__html: articleActions }}
```

Comment:
this should allow passing in a JSX child as well, like with `x-teaser`'s [`CustomSlot`](https://github.com/Financial-Times/x-dash/blob/b196dc20a09b861fce69e4e50fc77f3105f948b7/components/x-teaser/src/CustomSlot.jsx)

Reply:
actually, should this just pass `articleActions` into the teaser as the custom slot?

Reply:
I've changed it to allow passing a JSX child or a function now.

I didn't use the teaser custom slot, because there didn't seem to be any advantage to doing so, and it made the CSS simpler to just treat the teaser and actions slot as separate grid areas.